eth-rpc/server/serverdispatch.nim

```diff
@@ -20,7 +20,7 @@ proc processMessage(server: RpcServer, client: AsyncSocket, line: string) {.asyn
     if not server.procs.hasKey(methodName):
       await client.sendError(METHOD_NOT_FOUND, "Method not found", id, %(methodName & " is not a registered method."))
     else:
-      let callRes = server.procs[methodName](node["params"])
+      let callRes = await server.procs[methodName](node["params"])
```

Comment:
If a bogus JSON is received, the dictionary and field accesses here will raise an exception, right? Is the rest of the code prepared for this? Perhaps we should try to enumerate the possible exceptions returned by the JSON validation and marshaling code.

Reply:
I agree we need to enumerate exceptions and reduce them to the smallest quantity possible.

However, there's a check for the method name key just above it so this shouldn't cause a problem, should it?

Reply:
Extra checking added.